### ISSUE TITLE
Factorio update bug

### ISSUE THREAD

**Issue by dgibbs64** [GameServerManagers/LinuxGSM#2733]
from https://github.com/GameServerManagers/LinuxGSM/pull/2215#issuecomment-469864870

don't know how to a PR but here is my simple fix for this.

The error I talked about it a migrations error.
If you don't clear the tmpdir you will always cp the migrations from older versions.
Older servers that have files from 0.12.0 this ends into startup errors.
To fix this issues you can simply add a "tmpdir clear" to the update_factorio.sh
for me, this works.

I addes this function to the file:
```
fn_update_factorio_clear_tmp(){
	echo -e "Clear tmpdir...\c"
	fn_script_log "Directory ${tmpdir} cleared"
	rm -r ${tmpdir}/*
	local exitcode=$?
	if [ "${exitcode}" == "0" ]; then
		fn_print_ok_eol_nl
	else
		fn_print_fail_eol_nl
	fi
}
```
And made changes to fn_update_factorio_compare
```
fn_update_factorio_compare(){
	# Removes dots so if statement can compare version numbers.
	currentbuilddigit=$(echo "${currentbuild}" | tr -cd '[:digit:]')
	availablebuilddigit=$(echo "${availablebuild}" | tr -cd '[:digit:]')

	if [ "${currentbuilddigit}" -ne "${availablebuilddigit}" ]; then
		echo -e "\n"
		echo -e "Update available:"
		sleep 0.5
		echo -e "	Current build: ${red}${currentbuild} ${factorioarch} ${branch} ${default}"
		echo -e "	Available build: ${green}${availablebuild} ${factorioarch} ${branch}${default}"
		echo -e ""
		sleep 0.5
		echo -en "Applying update.\r"
		sleep 1
		echo -en "Applying update..\r"
		sleep 1
		echo -en "Applying update...\r"
		sleep 1
		echo -en "\n"
		fn_script_log "Update available"
		fn_script_log "Current build: ${currentbuild} ${factorioarch}${branch}"
		fn_script_log "Available build: ${availablebuild} ${factorioarch}${branch}"
		fn_script_log "${currentbuild} > ${availablebuild}"

		unset updateonstart

		check_status.sh
		if [ "${status}" == "0" ]; then
			fn_update_factorio_clear_tmp
			exitbypass=1
			fn_update_factorio_dl
			exitbypass=1
			command_start.sh
			exitbypass=1
			command_stop.sh
		else
			exitbypass=1
			command_stop.sh
			exitbypass=1
			fn_update_factorio_clear_tmp
			exitbypass=1
			fn_update_factorio_dl
			exitbypass=1
			command_start.sh
		fi
		sleep 2
		alert="update"
		alert.sh
	else
		echo -e "\n"
		echo -e "No update available:"
		echo -e "	Current build: ${green}${currentbuild} ${factorioarch} ${branch}${default}"
		echo -e "	Available build: ${green}${availablebuild} ${factorioarch} ${branch}${default}"
		echo -e ""
		fn_print_ok_nl "No update available"
		fn_script_log_info "Current build: ${currentbuild} ${factorioarch} ${branch}"
		fn_script_log_info "Available build: ${availablebuild} ${factorioarch} ${branch}"
	fi
}
```
The result ist, that the script clears the tmpdir before update the game.

**Comment by dgibbs64** [comment]
So after investigating the updater already removes the tmp directory. I have moved it so that no matter if it passes or failed it will always delete the tmp dir

**Comment by h3o66** [comment]
~Closed
bug fix in version 20.3.0 included
